client/src/components/VersionSaveComponent.tsx

```diff
@@ -1,13 +1,66 @@
-import { Button, Col, Input, Row } from 'antd'
+import { Button, Col, Dropdown, Input, Menu, Row } from 'antd'
 import React, { useState } from 'react'
-import { GetAnswerDocument, useSaveVersionMutation } from '../generated/graphql'
+import {
+  FileContextType,
+  GetAnswerDocument,
+  ListFilesDocument,
+  useChangeVersionMutation,
+  useGetAnswerQuery,
+  useSaveVersionMutation
+} from '../generated/graphql'
 import { messageService } from '../services/message'
 
 const VersionSaveComponent: React.FC<{
   answerId: string
-  commitMessage: string
-  versionId: string
 }> = props => {
+  const result = useGetAnswerQuery({
+    variables: { id: props.answerId }
+  })
+
+  const [changeVersion] = useChangeVersionMutation({
+    onCompleted: ({ changeVersion: didVersionChange }) => {
+      if (didVersionChange) {
+        messageService.success('Version got changed')
+      } else {
+        messageService.error('Selected version is the current version')
+      }
+      result.refetch()
+    },
+    refetchQueries: [
+      {
+        query: ListFilesDocument,
+        variables: {
+          context: {
+            id: props.answerId,
+            type: FileContextType.Answer
+          },
+          path: '/'
+        }
+      }
+    ]
+  })
+
+  const menuItemList = result.data?.answer.versions.map(commit => (
+    <Menu.Item
+      key={commit.versionKey}
+      onClick={() =>
+        changeVersion({
+          variables: { id: props.answerId, versionID: commit.versionKey }
+        }).then(() => result.refetch())
```

Comment:
Since we already use `refetchQueries` for the Mutation wenn can refetch this query there as well